Fix unknown color (Alpha), optimized get_gcolor_text using an array.

ron064 opened the pull request:
Known issues:
1. GColorBlack, GColorClear currently treated as same in replace_gbitmap_color
2. Didn't update gbitmap_fill_all_except
3. gbitmap_fill_all_except should probably skip GColorClear.

ron064 committed:
Fix unknown color (Alpha), optimized GColorText using an array.

Known issues:
1. GColorBlack, GColorClear currently treated as same in replace_gbitmap_color
2. Didn't update gbitmap_fill_all_except
3. gbitmap_fill_all_except should probably skip GColorClear.

rebootsramblings commented:
Ron thanks for the changes I will be testing them all soon. Can you check to see if you were using the latest files? There is missing some conditionals for the APP_LOGS and perhaps some other things. Would you mind taking a look before I merge?

ron064 commented:
Yes, I worked over the latest from main branch. If you look at the commit tab and click on the commit, you will see every line I touched and it is mostly the relevant lines. I updated include lines as well to keep convention that <> looks in system folders and "" look at local folder first.
As for APP_LOG, IMHO you should either put them in ifdef block, or redefine APP_LOG macro to disable it. when not debugging it is better the debug code will not compile (and not take space)

I updated the returning value to const char *, you may want to update your other text function in the same way, as it's a value that user is not expected to change.

When you publish library, there are few small subjects to consider that you don't care when it's your code. Occasionally I notice another item of this category.

rebootsramblings commented:
Ron, your help and explanations are very appreciated. I am still pretty new to publishing libraries so I appreciate the help and feedback.
